test.lua

```diff
@@ -5807,6 +5807,23 @@ function nntest.Cosine()
    mytester:assertTensorEq(cosine.gradWeight, cosine2.gradWeight, 0.000001, "Cosine gradWeight 2D err")
 end
 
+function nntest.ErrorHandling()
+    local l = nn.Linear(1, 1)
+    local p = nn.Parallel(1, 1):add(l)
+    local c = nn.Concat(1):add(p)
+    local model = nn.Sequential():add(nn.Identity()):add(c):add(nn.Identity())
+    local ok, err = pcall(function()
+        model:forward(torch.randn(1,2,2))
+    end)
+    mytester:assert(not ok, "Function call was expected to fail")
```

Comment:
It's minor but you might be able to use a single assert all in all by leveraging [assertErrorObj](https://github.com/torch/torch7/blob/master/doc/tester.md#asserterrorobjf-errcomp--message):

```Lua
mytester:assertErrorObj(
    function()
        model:forward(torch.randn(1,2,2))
    end,
    function(err)
        return err:find(expected_err) and err:find('size mismatch')
    end,
    "Failure expected or bad error message (missing information or reason)"
)
```